src/library/cratehighlightdelegate.cpp

```diff
@@ -0,0 +1,28 @@
+#include "library/cratehighlightdelegate.h"
+#include "library/treeitem.h"
+#include "trackinfoobject.h"
+#include "libraryfeature.h"
+#include "cratefeature.h"
+
+
+void CrateHighlightDelegate::initStyleOption(QStyleOptionViewItem *option,
+                     const QModelIndex &index) const{
+    QStyledItemDelegate::initStyleOption(option, index);
+    QStyleOptionViewItemV4 *optionV4 = qstyleoption_cast<QStyleOptionViewItemV4*>(option);
+
+    TreeItem *item = static_cast<TreeItem*>(index.internalPointer());
```

Comment:
I got this from TreeItemModel::data(), where it's done the same way. But i've added a condition for index.isValid() before. Thats the way we are doing it there too.